volume_type_validation/volume_type_validation.py

```diff
@@ -0,0 +1,240 @@
+                "recommendations": self._recommendations(host.id)
+            }
+
+        self.result.set_hosts(hosts)
+
+    def _find_invalid_in_groups(self, source_id, group_ids):
+        lives_on_vm = self.container_nodes.lives_on_type == 'Vm'
+        container_nodes_group = \
+            self.container_nodes.loc[group_ids][lives_on_vm]
+
+        container_nodes_taggings = self.container_nodes_tags
+
+        type_taggings = \
+            container_nodes_taggings[container_nodes_taggings['name'] ==
+                                     "type"].copy()
+
+        if not type_taggings.empty:
+            # TODO For master nodes, it can happen
+            # that the nodes
+            # running etcd have io1 and the rest
+            # gp2. Lets ignore these for now, until we can
+            # identify nodes running etcd
+            non_master_nodes_ids = \
+                type_taggings[type_taggings['value'] !=
+                              "master"]['container_node_id']
+            self._find_invalid(
+                source_id,
+                container_nodes_group[container_nodes_group['id'].isin(non_master_nodes_ids)]
+            )
+        else:
+            self._find_invalid(source_id, container_nodes_group)
+
+    def _find_invalid(self, source_id, container_nodes_group):
+        # Select container nodes of 1 group, but only those deployed on Vm
+        vms_ids = container_nodes_group['lives_on_id']
+
+        # Select volume attachments of these container nodes
+        group_volume_attachments = \
+            self.volume_attachments[self.volume_attachments['vm_id'].isin(vms_ids)]
+        group_volume_attachments_ids = group_volume_attachments["volume_id"]
+
+        # Select volumes of these volume attachments
+        group_volumes = \
+            self.volumes[self.volumes['id'].isin(group_volume_attachments_ids)]
+
+        # Asserts
+        self._find_invalid_volume_types(source_id, group_volumes, group_volume_attachments, container_nodes_group)
+
+    def _find_invalid_volume_types(
+            self,
+            source_id,
+            group_volumes,
+            group_volume_attachments,
+            container_nodes_group
+    ):
+        # volume_type_id of the whole group of volumes should be the same, if not add error message
+        if len(group_volumes['volume_type_id'].unique()) > 1:
+            # group_volumes.groupby('volume_type_id').groups
+
+            # Find what is recommended volume type, it will be the one used by majority of nodes
+            volume_type_groups = group_volumes.groupby('volume_type_id').groups
+            volume_type_groups_sizes = \
+                [[key, len(group)] for key, group in volume_type_groups.items()]
+            recommended_volume_type, recommended_volume_type_size = \
+                max(volume_type_groups_sizes, key=operator.itemgetter(1))
+            same_sizes = [[key, value] for key, value in volume_type_groups_sizes if
```

Comment:
the `value` is never used as property of the `same_sizes`, especially it's intentionally ignored on line 150, this can be simplified to:
```suggestion
            same_sizes = [key for key, value in volume_type_groups_sizes if value == recommended_volume_type_size]
```

Reply:
Removed `value` and `_count` down below
Addressed in 8fb9d5ad5fc223986f95be62e3f447f958f5d9f6